membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -244,4 +252,10 @@ object OptionEither {
   def liftFutureEither[A](x: Option[A]): OptionT[FutureEither, A] =
     apply(Future.successful(\/.right[String,Option[A]](x)))
 
+  def liftEither[A](x: Future[Option[A]])(implicit ex: ExecutionContext): OptionT[FutureEither, Option[A]] = {
```

Comment:
Still not sure of the best approach so that alertText can be None without halting the for comprehension, but this works in CODE. Suggestions very very welcome. Whenever I try to wrangle the types right in the for comprehension I can't get it to compile. 😬

Reply:
looks fine for this approach, and it should work.  Splitting it out is a good approach, as for comprehensions should just be (in throey) creating a data structure showing how each part of the program depends on the rest!

In terms of polish, we don't care that it's already an option, A is fine, also we are lifting the value inside the future into EitherOption not just Either so the name could be tweaked too:
```scala
def liftEitherOption[A](future: Future[A])(implicit ex: ExecutionContext): OptionT[FutureEither, A] = {
  apply(future map { value: A =>
    \/.right[String, Option[A]](Some(value))
  })
}
```

Reply:
Good point! Thanks.